test/suite/test_hs06.py

```diff
@@ -505,9 +505,14 @@ def test_hs_rec_modify(self):
 
         # Apply another update and evict the pages with the modifies out of cache.
         for i in range(1, 10000):
-            self.session.begin_transaction()
-            cursor[self.create_key(i)] = value2
-            self.session.commit_transaction('commit_timestamp=' + timestamp_str(6))
+            try:
+                self.session.begin_transaction()
+                cursor[self.create_key(i)] = value2
+                self.session.commit_transaction('commit_timestamp=' + timestamp_str(6))
+            except:
+                # We may have got a WT_ROLLBACK when attempted update the value. As all we're trying
+                # to do is evict pages from cache we don't need to retry this update.
+                self.session.rollback_transaction()
```

Comment:
It would be good to add a counter and assert that not *all* of the updates failed.